internal/provider/vm/vm_affinity_rule_schema.go

```diff
@@ -60,135 +59,99 @@ func vmAffinityRuleSchema() superschema.Schema {
-					Computed:            true,
+				},
+				Resource: &schemaR.StringAttribute{
+					Computed: true,
 					PlanModifiers: []planmodifier.String{
 						stringplanmodifier.UseStateForUnknown(),
 					},
 				},
+				DataSource: &schemaD.StringAttribute{
+					MarkdownDescription: " Required if `name` is not set.",
+					Optional:            true,
+					Validators: []validator.String{
+						stringvalidator.ExactlyOneOf(path.MatchRoot("name"), path.MatchRoot("id")),
+					},
+				},
 			},
-			"vdc": superschema.StringAttribute{
-				Resource: &v,
-			},
+			"vdc": vdc.SuperSchema(),
 			"name": superschema.StringAttribute{
-				Resource: &schemaR.StringAttribute{
-					Required:            true,
+				Common: &schemaR.StringAttribute{
 					MarkdownDescription: "VM affinity rule name.",
 				},
+				Resource: &schemaR.StringAttribute{
+					Required: true,
+				},
+				DataSource: &schemaD.StringAttribute{
+					MarkdownDescription: " Required if `id` is not set.",
+					Optional:            true,
+					Validators: []validator.String{
+						stringvalidator.ExactlyOneOf(path.MatchRoot("name"), path.MatchRoot("id")),
+					},
+				},
 			},
 			"polarity": superschema.StringAttribute{
-				Resource: &schemaR.StringAttribute{
-					Required:            true,
+				Common: &schemaR.StringAttribute{
 					MarkdownDescription: fmt.Sprintf("One of `%s`, `%s`", govcdtypes.PolarityAffinity, govcdtypes.PolarityAntiAffinity),
+				},
+				Resource: &schemaR.StringAttribute{
+					Required: true,
 					PlanModifiers: []planmodifier.String{
 						stringplanmodifier.RequiresReplace(),
 					},
 					Validators: []validator.String{
 						stringvalidator.OneOf(govcdtypes.PolarityAffinity, govcdtypes.PolarityAntiAffinity),
 					},
 				},
+				DataSource: &schemaD.StringAttribute{
+					Optional: true,
+				},
 			},
 			"enabled": superschema.BoolAttribute{
-				Resource: &schemaR.BoolAttribute{
-					Optional:            true,
-					Computed:            true,
-					Default:             booldefault.StaticBool(true),
+				Common: &schemaR.BoolAttribute{
 					MarkdownDescription: "`True` if this affinity rule is enabled. Default is `true`.",
 				},
-			},
-			"required": superschema.BoolAttribute{
 				Resource: &schemaR.BoolAttribute{
 					Optional: true,
 					Computed: true,
 					Default:  booldefault.StaticBool(true),
+				},
+				DataSource: &schemaD.BoolAttribute{
+					Optional: true,
+				},
+			},
+			"required": superschema.BoolAttribute{
+				Common: &schemaR.BoolAttribute{
 					MarkdownDescription: "`True` if this affinity rule is required. When a rule is mandatory, " +
 						"a host failover will not power on the VM if doing so would violate the rule. Default is `true`.",
 				},
+				Resource: &schemaR.BoolAttribute{
+					Optional: true,
+					Computed: true,
+					Default:  booldefault.StaticBool(true),
+				},
+				DataSource: &schemaD.BoolAttribute{
+					Optional: true,
+				},
 			},
```

Comment:
The description < Default is `true` > is only in Resource 

In datasource this attribute is realy `Optional` ? Not Compute ?